cmd/publishing-bot/config/config.go

```diff
@@ -51,4 +51,7 @@ type Config struct {
 	// BasePublishScriptPath determine the base path where we will look for a
 	// publishing scripts in the source repo. It defaults to ./publishing_scripts'.
 	BasePublishScriptPath string `yaml:"base-publish-script-path,omitempty"`
+
+	// name of the default git branch in the repo. defaults to master
+	GitDefaultBranch string `yaml:"git-default-branch,omitempty"`
```

Comment:
Should we make this a pointer since it's optional?

Reply:
There are other fields also in the config struct, which are optional and kept as the type itself, without making it a pointer.  I think having a pointer helps when we are later displaying the struct to the user, like in case of k8s APIs. Does it really make sense here? 
Also, it will involve checking and dereferencing everywhere in code where we use it.  I think using pointer when its a struct is ok, other wise dont think it really helps in our case.